crates/blockchain-tree/src/chain.rs

```diff
@@ -203,11 +205,17 @@ impl AppendableChain {
 
         let provider = BundleStateProvider::new(state_provider, bundle_state_data_provider);
 
-        let mut executor = externals.executor_factory.with_state(&provider);
+        let db = StateProviderDatabase::new(&provider);
+        let executor = externals.executor_factory.executor(db);
         let block_hash = block.hash();
         let block = block.unseal();
-        executor.execute_and_verify_receipt(&block, U256::MAX)?;
-        let bundle_state = executor.take_output_state();
+        let state = executor.execute((&block, U256::MAX).into()).map_err(Into::into)?;
```

Comment:
I'd love if we got rid of the `total_difficulty` param at some point as well, given it's pure debt other than eth pre-merge